components/core/App.js

```diff
@@ -1,27 +1,35 @@
-import { useEffect, useState } from 'react';
+import React, { useState } from 'react';
 
 // import data from json
 import { getAllDemos } from 'utils/data/data-access';
 
 // Demo component
 import { Demo } from '..';
+import { SearchBox } from './SearchBox';
 
 const App = () => {
   const [demos, setDemos] = useState([]);
   const [loading, setLoading] = useState(true);
 
-  useEffect(() => {
-    const allDemos = getAllDemos();
-    console.log({ allDemos });
-    setDemos(allDemos);
+  const searchAPI = (searchText) => {
+    const lowerSearchText = searchText.toLowerCase();
+    setLoading(true);
+    let searchedDemos = getAllDemos();
+    if (searchText) {
+      searchedDemos = searchedDemos
+        .filter(demo => demo.title?.toLowerCase().includes(lowerSearchText));
+    }
+
+    setDemos(searchedDemos);
     setLoading(false);
-  }, []);
+  };
 
   return (
     <>
+      <SearchBox onSearch={searchAPI} />
       <div className="demo-list">
         {(!loading && demos.length) > 0 &&
-          demos.map((demo, index) => <Demo demo={demo} key={index} />)}
+        demos.map((demo, index) => <Demo demo={demo} key={index} />)}
```

Comment:
@omriattiya Can we add a "No matched demo found" warning at the center of the page if it doesn't match criteria? Please feel free to make it as a separate component.